src/qt/optionsdialog.cpp

```diff
@@ -45,6 +45,8 @@ OptionsDialog::OptionsDialog(QWidget *parent, bool enableWallet) :
     ui->threadsScriptVerif->setMinimum(-GetNumCores());
     ui->threadsScriptVerif->setMaximum(MAX_SCRIPTCHECK_THREADS);
 
+    ui->networkPort->setValidator(new QIntValidator(1, 65535, this));
```

Comment:
We need a better protection to not run into a startup termination if the user did enter a invalid port. IIRC the `QIntValidator` still allows one to enter `0` as port number which will very likely result in a locked Bitcoin-QT (startup will stop because of a invalid port). Haven't tested although.
Maybe check: https://github.com/bitcoin/bitcoin/pull/7025